RecoTracker/TkSeedGenerator/plugins/DeepCoreSeedGenerator.cc

```diff
@@ -0,0 +1,609 @@
+
+              if (det == goodDet1 || det == goodDet3 || det == goodDet4 || det == globDet) {
+                fillPixelMatrix(aCluster, lay, localInter, det, input_tensors);
+              }
+            }  //cluster in ROI
+          }    //cluster
+        }      //detset
+
+        //here the NN produce the seed from the filled input
+        std::pair<double[jetDimX][jetDimY][Nover][Npar], double[jetDimX][jetDimY][Nover]> seedParamNN =
+            DeepCoreSeedGenerator::SeedEvaluation(input_tensors, output_names);
+
+        for (int i = 0; i < jetDimX; i++) {
+          for (int j = 0; j < jetDimY; j++) {
+            for (int o = 0; o < Nover; o++) {
+              if (seedParamNN.second[i][j][o] > (probThr - o * 0.1 - (l2off ? 0.35 : 0))) {
+                std::pair<bool, Basic3DVector<float>> interPair =
+                    findIntersection(bigClustDir, (reco::Candidate::Point)jetVertex.position(), globDet);
+                auto localInter = globDet->specificSurface().toLocal((GlobalPoint)interPair.second);
+
+                int flip = pixelFlipper(globDet);  // 1=not flip, -1=flip
+                int nx = i - jetDimX / 2;
+                int ny = j - jetDimY / 2;
+                nx = flip * nx;
+                std::pair<int, int> pixInter = local2Pixel(localInter.x(), localInter.y(), globDet);
+                nx = nx + pixInter.first;
+                ny = ny + pixInter.second;
+                LocalPoint xyLocal = pixel2Local(nx, ny, globDet);
+
+                double xx = xyLocal.x() + seedParamNN.first[i][j][o][0] * 0.01;  //0.01=internal normalization of NN
+                double yy = xyLocal.y() + seedParamNN.first[i][j][o][1] * 0.01;  //0.01=internal normalization of NN
+                LocalPoint localSeedPoint = LocalPoint(xx, yy, 0);
+
+                double track_eta =
+                    seedParamNN.first[i][j][o][2] * 0.01 + bigClustDir.eta();  //pay attention to this 0.01
+                double track_theta = 2 * std::atan(std::exp(-track_eta));
+                double track_phi =
+                    seedParamNN.first[i][j][o][3] * 0.01 + bigClustDir.phi();  //pay attention to this 0.01
+
+                double pt = 1. / seedParamNN.first[i][j][o][4];
+                double normdirR = pt / sin(track_theta);
+
+                const GlobalVector globSeedDir(
+                    GlobalVector::Polar(Geom::Theta<double>(track_theta), Geom::Phi<double>(track_phi), normdirR));
+                LocalVector localSeedDir = globDet->surface().toLocal(globSeedDir);
+                int64_t seedid = (int64_t(xx * 200.) << 0) + (int64_t(yy * 200.) << 16) +
+                                 (int64_t(track_eta * 400.) << 32) + (int64_t(track_phi * 400.) << 48);
+                if (ids.count(seedid) != 0) {
+                  continue;
+                }
+                if (true) {  //1 TO JET CORE; 0=NO JET CORE (seeding iteration skipped, useful to total eff and FakeRate comparison)
```

Comment:
remove the `if (true)` as it's redundant.

Reply:
removed the if, but added a comment to explain how to quickly disable jetcore seeding for testing purposes